x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/integration/DocumentLevelSecurityTests.java

```diff
@@ -963,125 +971,130 @@ public void testGlobalAggregation() throws Exception {
-    // List<StringTerms.Bucket> fruitBuckets = fruits.getBuckets();
-    // assertTrue(fruitBuckets.stream().anyMatch(bucket -> bucket.getKeyAsString().equals("apple") && bucket.getDocCount() == 1));
-    // assertTrue(fruitBuckets.stream().anyMatch(bucket -> bucket.getKeyAsString().equals("grape") && bucket.getDocCount() == 0));
-    // // counts
-    // LongTerms counts = response.getAggregations().get("counts");
-    // assertThat(counts.getBuckets().size(), equalTo(2));
-    // List<LongTerms.Bucket> countsBuckets = counts.getBuckets();
-    // assertTrue(countsBuckets.stream().anyMatch(bucket -> bucket.getKeyAsString().equals("-1") && bucket.getDocCount() == 1));
-    // assertTrue(countsBuckets.stream().anyMatch(bucket -> bucket.getKeyAsString().equals("-4") && bucket.getDocCount() == 0));
-    // // colors
-    // for (int i = 1; i <= 3; i++) {
-    // StringTerms colors = response.getAggregations().get("colors" + i);
-    // assertThat(colors.getBuckets().size(), equalTo(1));
-    // assertThat(colors.getBuckets().get(0).getKeyAsString(), equalTo("red"));
-    // assertThat(colors.getBuckets().get(0).getDocCount(), equalTo(1L));
-    // }
-    // // nested
-    // StringTerms nested = response.getAggregations().get("nested");
-    // assertThat(nested.getBuckets().size(), equalTo(1));
-    // assertThat(nested.getBuckets().get(0).getKeyAsString(), equalTo("red"));
-    // assertThat(nested.getBuckets().get(0).getDocCount(), equalTo(1L));
-    // StringTerms innerFruits = nested.getBuckets().get(0).getAggregations().get("fruits");
-    // List<StringTerms.Bucket> innerFruitsBuckets = innerFruits.getBuckets();
-    // assertTrue(innerFruitsBuckets.stream().anyMatch(b -> b.getKeyAsString().equals("apple") && b.getDocCount() == 1));
-    // assertTrue(innerFruitsBuckets.stream().anyMatch(b -> b.getKeyAsString().equals("grape") && b.getDocCount() == 0));
-    // assertThat(innerFruitsBuckets.size(), equalTo(2));
-    //
-    // for (int i = 0; i <= 1; i++) {
-    // String parentBucketKey = innerFruitsBuckets.get(i).getKeyAsString();
-    // LongTerms innerCounts = innerFruitsBuckets.get(i).getAggregations().get("counts");
-    // assertThat(innerCounts.getBuckets().size(), equalTo(2));
-    // List<LongTerms.Bucket> icb = innerCounts.getBuckets();
-    // if ("apple".equals(parentBucketKey)) {
-    // assertTrue(icb.stream().anyMatch(bucket -> bucket.getKeyAsString().equals("-1") && bucket.getDocCount() == 1));
-    // } else {
-    // assertTrue(icb.stream().anyMatch(bucket -> bucket.getKeyAsString().equals("-1") && bucket.getDocCount() == 0));
-    // }
-    // assertTrue(icb.stream().anyMatch(bucket -> bucket.getKeyAsString().equals("-4") && bucket.getDocCount() == 0));
-    // }
-    // }
-    // );
-    // }
-    //
-    // private static IndicesAdminClient indicesAdmin() {
-    // return admin().indices();
-    // }
+    public void testZeroMinDocAggregation() throws Exception {
```

Comment:
this is really hard to diff... here is a better diff (unzip and open in browser):

[DocumentLevelSecurityTests_diff.html.zip](https://github.com/elastic/elasticsearch/files/15266822/DocumentLevelSecurityTests_diff.html.zip)

gitdiff upstream/main -- x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/integration/DocumentLevelSecurityTests.java and using beyond compare to generate html report